kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -792,90 +810,97 @@ class ResidualBasedBlockBuilderAndSolver
      * should refer to the particular Builder And Solver choosen
      * @param pScheme The integration scheme considered
      * @param rModelPart The model part of the problem to solve
-     * @param A The LHS matrix
-     * @param Dx The Unknowns vector
-     * @param b The RHS vector
+     * @param rA The LHS matrix
+     * @param rDx The Unknowns vector
+     * @param rb The RHS vector
      */
     void ApplyDirichletConditions(
         typename TSchemeType::Pointer pScheme,
         ModelPart& rModelPart,
-        TSystemMatrixType& A,
-        TSystemVectorType& Dx,
-        TSystemVectorType& b) override
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
+        ) override
     {
-        std::size_t system_size = A.size1();
-        std::vector<double> scaling_factors (system_size, 0.0);
+        std::size_t system_size = rA.size1();
+        std::vector<double> scaling_factors (system_size, 1.0);

```

Comment:
not your fault, but here it would be better to use a Vector instead of an std::vector.

the reason for which i am telling is thatt otherwise initialization does not happen in parallel, which also implies that first touching does not work correctly